### PR TITLE
Upgrade to ember 2.12.3

### PR DESCRIPTION
**Pull request by dafortin**
**This project uses [semver](http://semver.org), please check the scope of this pr:**

- [ ] #none# - documentation fixes and/or test additions
- [x] #patch# - backwards-compatible bug fix
- [ ] #minor# - adding functionality in a backwards-compatible manner
- [ ] #major# - incompatible API change

# CHANGELOG
* **Upgrade** to ember 2.12.3

### CODE REVIEW

**Comment by notmessenger** [comment]
- _config/environment.js_ should have `/* eslint-env node */` added to it
- _ember-cli-build.js_ should have `/* eslint-env node */` added to it
- _index.js_ should have `/* eslint-env node */` added to it
- _testem.js_ should have `/* eslint-env node */` added to it
- _tests/dummy/config/environment.js_ should have `/* eslint-env node */` added to it
_ The _tests/.eslintrc.js_ files should have modifications and does not

- should set the `broccoli-asset-rev` dependency in _package.json_ to `^2.4.5` so that we're in alignment with the value from `ember-cli@2.12.3` and the other repos being set to the same
- the `ember-cli-test-loader` dependency should be removed from _package.json_
- should set the `ember-export-application-global` dependency in _package.json_ to `^1.0.5` so that we're in alignment with the value from `ember-cli@2.12.3` and the other repos being set to the same
- should set the `ember-load-initializers` dependency in _package.json_ to `^0.6.0` so that we're in alignment with the value from `ember-cli@2.12.3` and the other repos being set to the same

File: package.json [hunk 1]
```diff
@@ -89,14 +88,15 @@
   },
   "dependencies": {
     "broccoli-autoprefixer": "^4.1.0",
+    "broccoli-babel-transpiler": "^5.0.0",
```
Comment (dafortin) [inline package.json:90]: @juwara0 @notmessenger  Should the version be bump by a major because of this?

**Comment by notmessenger** [comment]
@dafortin  What is the `.ember-try` entry you added to the _.gitignore_ referring to?  I do not see any such file in the repo, nor does https://github.com/ember-cli/ember-try make any reference to such a file either.

**Comment by dafortin** [comment]
I'm getting a lot of pollution in my ide with .node_modules.ember-try files and I was not getting this before. Now I'm getting this for core and other repos.

**Comment by notmessenger** [comment]
@dafortin We should not be adding project-level entries to the _.gitignore_ file that cater only to a specific developer's setup.  You should instead leverage the hierarchical nature of the application of _.gitignore_ rules (https://git-scm.com/docs/gitignore#_description) and ignore this file in your own environment instead.

Thoughts @juwara0?

**Comment by dafortin** [comment]
I was expecting that you might experiment the same in your env. If it's not the case then I can move it in my personal stuff.

**Comment by dafortin** [comment]
Let's wait to 2017.06.26 to merge this

File: index.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-env node */
 /* globals module */
```
Comment (notmessenger) [inline index.js]: Is this needed?

**Comment by sglanzer-fire** [comment]
"Let's wait to 2017.06.26 to merge this"

I'd suggest waiting until after June 30th to be safe

**Comment by dafortin** [comment]
@juwara0 @notmessenger I think it should be ready to merge. Feel free to merge if you are ok with this PR.

File: config/ember-try.js [hunk 3]
```diff
@@ -1,129 +1,41 @@
+/* eslint-env node */
```
Comment (notmessenger) [inline config/ember-try.js:1]: This file is not the same structure as being used in all other related ember cli upgrade PRs

File: ember-cli-build.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
+/* eslint-env node */
 /* global require, module */
-var EmberApp = require('ember-cli/lib/broccoli/ember-addon')
+const EmberApp = require('ember-cli/lib/broccoli/ember-addon')
```
Comment (notmessenger) [inline ember-cli-build.js]: Instances of `EmberApp` should be `EmberAddon`

File: index.js (hunk 2 above)
Comment (notmessenger) [inline index.js]: Remove this line

File: package.json [hunk 5]
```diff
@@ -89,14 +88,15 @@
...
     "broccoli-file-creator": "^1.1.0",
     "broccoli-funnel": "^1.0.1",
     "broccoli-merge-trees": "^1.0.0",
     "broccoli-svgstore": "git://github.com/ciena-frost/broccoli-svgstore.git",
     "chalk": "^1.1.3",
     "child-process-promise": "^2.2.1",
     "ember-cli-babel": "^5.1.6",
```
Comment (notmessenger) [inline package.json]: Should be `^5.1.7`

File: package.json [hunk 6]
```diff
@@ -46,30 +46,29 @@
...
-    "ember-cli-test-loader": "^1.1.1",
+    "ember-cli-shims": "^1.0.2",
     "ember-cli-uglify": "^1.2.0",
     "ember-code-snippet": "1.7.0",
     "ember-computed-decorators": "~0.3.0",
     "ember-concurrency": "~0.7.15",
-    "ember-data": "^2.11.0",
+    "ember-data": "~2.12.0",
     "ember-disable-prototype-extensions": "1.1.0",
     "ember-disable-proxy-controllers": "^1.0.1",
```
Comment (notmessenger) [inline package.json]: Should be removed

File: package.json [hunk 7]
```diff
@@ -46,30 +46,29 @@
...
-    "ember-cli-shims": "^1.0.2",
     "ember-cli-notifications": "^4.1.6",
-    "ember-cli-test-loader": "^1.1.1",
+    "ember-cli-shims": "^1.0.2",
     "ember-cli-uglify": "^1.2.0",
     "ember-code-snippet": "1.7.0",
     "ember-computed-decorators": "~0.3.0",
     "ember-concurrency": "~0.7.15",
-    "ember-data": "^2.11.0",
+    "ember-data": "~2.12.0",
```
Comment (notmessenger) [inline package.json]: Is this version compatible with `ember-cli-mirage`?  `ember-cli-mirage` is the package that requires this dev dependency so shouldn't we be using whatever version of ember data version `0.2.5` of `ember-cli-mirage` installs?

File: tests/dummy/config/environment.js [hunk 8]
```diff
@@ -1,3 +1,4 @@
+/* eslint-env node */
 module.exports = function (environment) {
```
Comment (notmessenger) [inline tests/dummy/config/environment.js:2]: `ENV.rootURL = '/'` in the test configuration should be removed

File: package.json (hunk 7 above)
Comment (dafortin) [reply package.json]: It's using `^2.10.0` so we can probably use the same version.

**Comment by dafortin** [comment]
@notmessenger I handled all the PR comments

**Comment by notmessenger** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/ciena-frost/ember-frost-core/pull-request/467/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ciena-frost/ember-frost-core)

**Pull request merged**